src/os_crypto/shared/keys.c

```diff
@@ -42,7 +42,7 @@ static void move_netdata(keystore *keys, const keystore *old_keys)
             memcpy(&keys->keyentries[keyid]->peer_info, &old_keys->keyentries[i]->peer_info, sizeof(struct sockaddr_in));
 
             snprintf(strsock, sizeof(strsock), "%d", keys->keyentries[keyid]->sock);
-            OSHash_Add(keys->keyhash_sock, strsock, keys->keyentries[keyid]);
+            rbtree_insert(keys->keytree_sock, strsock, keys->keyentries[keyid]);
```

Comment:
Use the mutex `keytree_sock_mutex`.

Reply:
`move_netdata()` is part of `UpdateKeys()`, which assumes that the global lock is acquired.